### ISSUE TITLE
ttnn.rpow_bw computes the gradient of x**base instead of base**x; NaN for every negative input

### ISSUE THREAD

**Issue by ping-dev-ui** [tenstorrent/tt-metal#55289]
**Component / Area:** ttnn eltwise unary backward (composite), `rpow_bw`

**Issue Type:** Bad Outputs

## Summary

`ttnn.rpow(x, base)` is `base ** x`: the SFPU kernel says so
([ckernel_sfpu_rpow.h#L12](https://github.com/tenstorrent/tt-metal/blob/9fa00fce13cda1538f0130695576408ff5b05665/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_rpow.h#L12),
`// ttnn.rpow(exponent, scalar_base) = pow(scalar_base, exponent)`) and the forward golden is `torch.pow(dim, input_tensor_a)`
([unary.py#L812-L818](https://github.com/tenstorrent/tt-metal/blob/9fa00fce13cda1538f0130695576408ff5b05665/ttnn/ttnn/operations/unary.py#L812-L818)).

`ttnn.rpow_bw` returns `grad * base * x**(base-1)`, which is the derivative of `x ** base`
([unary_backward.cpp#L802-L817](https://github.com/tenstorrent/tt-metal/blob/9fa00fce13cda1538f0130695576408ff5b05665/ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp#L802-L817)).
The correct derivative of `base ** x` is `base**x * ln(base)`. The two agree only at `x == 1`. Because `x**(base-1)` is
undefined for negative `x` with a non-integer `base`, the op also forces NaN for every `x < 0`, where `base ** x` is an
ordinary finite number.

```cpp
// unary_backward.cpp L802-817 (paraphrased)
grad_result = zeros_like(input);
if (exponent != 0) {
    grad_result = grad * (pow(input, exponent - 1) * exponent);   // d/dx x^exponent
    grad_result = where(ltz(input), NaN, grad_result);
}
```

This is a source-level analysis with a host replay; I do not have a card. The host replay reproduces the composition in
float32 torch; the device numbers will differ only by the usual ULP-level SFPU error, not in kind.

## Observed vs expected (grad = 1, fp32)

| base | x | `rpow_bw` now | correct `base**x * ln(base)` |
|---|---|---|---|
| 2.0 | -1.0 | **NaN** | 0.3466 |
| 2.0 | 0.0 | 0 | 0.6931 |
| 2.0 | 2.0 | 4.0 | 2.7726 |
| 2.0 | 3.0 | 6.0 | 5.5452 |
| 5.7 | 3.0 | 996.2 | 322.3 |
| 15.2 | 2.0 | 286068 | 628.7 |
| 15.2 | 0.5 | 0.0008 | 10.61 |

## Why the tests do not catch it

The registered backward golden replicates the same misreading:
[unary_backward.py#L197-L202](https://github.com/tenstorrent/tt-metal/blob/9fa00fce13cda1538f0130695576408ff5b05665/ttnn/ttnn/operations/unary_backward.py#L197-L202)
calls `_golden_function_unary_backward_with_float("pow", grad, input, alpha)`, i.e. autograd of `torch.pow(input, alpha)`
(`input ** alpha`). The nightly test
([test_backward_rpow.py](https://github.com/tenstorrent/tt-metal/blob/9fa00fce13cda1538f0130695576408ff5b05665/tests/ttnn/nightly/unit_tests/operations/eltwise/backward/test_backward_rpow.py))
and the sweep golden compare only against that, so the op and its golden agree with each other and both disagree with the
forward op's own golden.

## Reproducer (host only, `pip install torch`)

```python
import torch

def ttnn_rpow_bw(grad, x, base):            # unary_backward.cpp L802-817
    if base == 0.0:
        return torch.zeros_like(x)
    r = grad * (torch.pow(x, base - 1.0) * base)
    return torch.where(x < 0, torch.tensor(float("nan")), r)

x = torch.tensor([-2., -1., 0., 0.5, 1., 2., 3.])
g = torch.ones_like(x)
for base in (1.5, 2.0, 5.7, 15.2):
    xr = x.clone().requires_grad_(True)
    torch.pow(base, xr).backward(g)         # autograd of the forward op's golden, torch.pow(base, x)
    print(base, ttnn_rpow_bw(g, x, base).tolist(), xr.grad.tolist())
```

## Proposed fix

```cpp
// d/dx base^x = base^x * ln(base). Forward docs already state non-positive bases are unsupported.
Tensor result = ttnn::multiply(grad, ttnn::rpow(input, exponent, output_mem_config), std::nullopt, output_mem_config);
result = ttnn::multiply(result, std::log(exponent), std::nullopt, output_mem_config);
```

Keep the `exponent == 0` zeros path (matches `torch.pow(0., x)` autograd for `x > 0`). Change the backward golden and the
sweep golden to autograd of `torch.pow(alpha, input)`, and add `x < 0` and `x == 0` rows to the nightly test so the
regression is pinned.

## Acceptance criteria (proposed)

- `ttnn.rpow_bw` matches autograd of `torch.pow(base, x)` to the existing PCC tolerance for `x` in `[-100, 100]` and
  bases `{1.5, 2.0, 5.7, 15.2}`, including negative `x`.
- Golden, sweep and nightly test updated to the corrected reference.

**Comment by harmonygemini8-collab** [comment]
Requesting assignment.

🔧 Plan: Replace the `x**base` derivative (`exponent * pow(input, exponent-1)` + ltz NaN) with the correct `d/dx base**x = base**x · ln(base)`: compose `grad * rpow_forward(input, base) * log(base)` from existing ops. Edge semantics per torch: base==1 → 0 everywhere (ln 1 = 0); base<=0 → NaN (ln undefined); negative x is now naturally valid since base**x > 0 for base>0 — removes the blanket ltz(input) NaN. Golden tests vs torch autograd over base ∈ {0.5, 2, e, 1, -2} × mixed-sign x, incl. integer/non-integer x with negative base.

📂 Files: `unary_backward.cpp` rpow_bw (~L802-817) + tests.

⏱️ ETA: 3 days. 💳 Payment: USDC (Base)

I have read and agree to the [Bounty Program Terms](https://docs.tenstorrent.com/bounty_terms.html).

**Comment by ZacLou** [comment]
I have completed root-cause analysis for this issue. The fix: replace the `x**base` derivative (`grad * base * x**(base-1)`) with the correct `d/dx base**x = base**x · ln(base)` in unary_backward.cpp L802-L817, sync the Python golden test, and add regression coverage for negative inputs and base-edge values. Could you assign #55289 to me?

**Comment by ZacLou** [comment]
Opened a PR that implements the fix described above: #55302.

**Comment by ZacLou** [comment]
Hi maintainers, the fix is ready in PR #55302. Could you please:`n1. Trigger the CI / approve the workflows for this first-time contribution.`n2. Confirm the bounty payout method and currency for #55289.`n`nHappy to address any review feedback promptly.

**Comment by ping-dev-ui** [comment]
Reporter here. I have the fix ready with tests on my fork (https://github.com/ping-dev-ui/tt-metal/tree/fix/rpow-bw-gradient): the derivative becomes grad * rpow(input, exponent) * ln(exponent), the x < 0 -> NaN guard is removed, the Python golden and the sweep golden are corrected to differentiate torch.pow(exponent, input), and a nightly case covers negative and zero inputs. Holding the PR until assignment per the bounty terms. @jberkowitzTT could you assign this to me when it is reviewed?

**Comment by ZacLou** [comment]
I have read and agree to the [Tenstorrent Bounty Program Terms & Conditions](https://docs.tenstorrent.com/bounty_terms.html). PR #55302 is already open with the fix and regression tests. Could you please assign this issue to me so the bounty is tracked correctly? Thanks.

**Comment by ping-dev-ui** [comment]
I have read and agree to the Tenstorrent Bounty Program Terms & Conditions (https://docs.tenstorrent.com/bounty_terms.html).

**Comment by jberkowitzTT** [comment]
Hi @ping-dev-ui - please submit all bounty proposals to [bounties@tenstorrent.com](mailto:bounties@tenstorrent.com). If approved, we'll create the issue and assign it to you as a bounty

Thank you for your enthusiasm & understanding
